crypto/fipsmodule/evp/p_ec.c

```diff
@@ -226,6 +226,37 @@ static int pkey_ec_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2) {
   }
 }
 
+static int pkey_ec_ctrl_str(EVP_PKEY_CTX *ctx,
+                            const char *type, const char *value)
+{
+  if (strcmp(type, "ec_paramgen_curve") == 0) {
+    int nid;
+    nid = EC_curve_nist2nid(value);
+    if (nid == NID_undef)
+      nid = OBJ_sn2nid(value);
+    if (nid == NID_undef)
+      nid = OBJ_ln2nid(value);
+    if (nid == NID_undef) {
+      OPENSSL_PUT_ERROR(EVP, EC_R_WRONG_CURVE_PARAMETERS);
```

Comment:
```suggestion
      OPENSSL_PUT_ERROR(EVP, EC_R_INVALID_ENCODING);
```
or `EC_R_INVALID_FORM` might make more sense here.